src/C50A0.c

```diff
@@ -158,9 +192,55 @@ INCLUDE_ASM(s32, "C50A0", func_80132D94);
 
 INCLUDE_ASM(s32, "C50A0", render_item_entities);
 
-INCLUDE_ASM(s32, "C50A0", remove_item_entity_by_reference);
+void remove_item_entity_by_reference(ItemEntity* entity) {
+    s32 index;
+
+    for (index = 0; index < 0x100; index++) {
```

Comment:
0x100 == ARRAY_COUNT(D_801565A0) + 1?
Perhaps this would work:
```suggestion
    for (index = 0; index <= ARRAY_COUNT(D_801565A0); index++) {
```

It would be worth attaching a `/// @bug` comment to this function (reading past the end of an array); the off-by-one was likely a typo

Reply:
This is not actually out of bounds as it turns out.